asf_tools/composite.py

```diff
@@ -0,0 +1,306 @@
+    dst = gdal.Open(fi)
+    p1 = dst.GetProjection()
+    ptr = p1.find("UTM zone ")
+    if ptr != -1:
+        (zone, hemi) = [t(s) for t, s in zip((int, str), re.search("(\d+)(.)", p1[ptr:]).groups())]
+    return zone
+
+
+def parse_zones(files):
+    """Return the zone numbers of all files given"""
+    zones = []
+    for fi in files:
+        zone = get_zone_from_proj(fi)
+        if zone:
+            zones.append(zone)
+    return np.asarray(zones, dtype=np.int8)
+
+
+def reproject_to_median_utm(files, pol, resolution=None):
+    """Reproject a bunch of UTM geotiffs to the median UTM zone.
+       Use either the given resolution or the largest resolution in the stack"""
+
+    if len(files) < 2:
+        return None
+
+    # Set the pixel size
+    if resolution:
+        pix_size = resolution
+        logging.info(f"Changing pixel size to {pix_size}")
+    else:
+        pix_size = get_max_pixel_size(files)
+        logging.info(f"Using maximum pixel size {pix_size}")
+
+    # Get the median UTM zone and hemisphere
+    home_zone = np.median(parse_zones(files))
+    logging.info(f"Home zone is {home_zone}")
+    hemi = get_hemisphere(files[0])
+    logging.info(f"Hemisphere is {hemi}")
+
+    # Reproject files as needed
+    logging.info("Checking projections")
+    new_files = []
+    for fi in files:
+        fi_name = os.path.split(fi)[1]
+        my_zone = get_zone_from_proj(fi)
+        name = fi_name.replace(".tif", "_reproj.tif")
+        afi = fi.replace(f"_{pol}.tif", "_area.tif")
+        aname = fi_name.replace(f"_{pol}.tif", "_area_reproj.tif")
+        if not os.path.isfile(name):
+            x, y, trans, proj = saa.read_gdal_file_geo(saa.open_gdal_file(fi))
+            if my_zone != home_zone:
+                logging.info(f"Reprojecting {fi} to {name}")
+                if hemi == "N":
+                    proj = ('EPSG:326%02d' % int(home_zone))
+                else:
+                    proj = ('EPSG:327%02d' % int(home_zone))
+                gdal.Warp(name, fi, dstSRS=proj, xRes=pix_size, yRes=pix_size, targetAlignedPixels=True)
+                gdal.Warp(aname, afi, dstSRS=proj, xRes=pix_size, yRes=pix_size, targetAlignedPixels=True)
+                new_files.append(name)
+            elif x < pix_size:
+                # Need to reproject to desired resolution
+                logging.info(f"Changing resolution of {fi} to {pix_size}")
+                gdal.Warp(name, fi, xRes=pix_size, yRes=pix_size, targetAlignedPixels=True)
+                gdal.Warp(aname, afi, xRes=pix_size, yRes=pix_size, targetAlignedPixels=True)
+                new_files.append(name)
+            else:
+                logging.info(f"No reprojection needed; Linking {fi} to {name}")
+                os.symlink(fi, name)
+                os.symlink(afi, aname)
+                new_files.append(name)
+        else:
+            logging.info(f"Found previous reproj file {fi} - taking no action")
+            new_files.append(fi)
```

Comment:
I think this may need to be `new_files.append(name)`; I'm seeing all of my scenes smashed together a single UTM zone when using previously reprojected files.

Reply:
Also seeing reprojection issues when the working directory does not contain all of the original tiffs, e.g. `composite.py -i ../*/*_VV.tif`.  The current code may assume all the backscatter/area tiffs are sitting next to each other in the working directory when using the `-i` flag?

Reply:
Using previously projected files doesn't seem to work correctly.  I think the thing to do is to remove recognition of previous reproj files.  Instead, calculate them fresh every time.